src/hooks/screen/useChatScreenHook.ts

```diff
@@ -95,10 +98,36 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
       if (type === 'ANONYMOUS') {
         response = await AnonymousMessageRepo.sendAnonymousMessage(selectedChannel?.id, message);
       } else {
+        const allAttachmentPromises: Promise<void>[] = attachments.map(async (item: any) => {
+          if (item.type === 'image') {
+            const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+            return {
+              ...item,
+              asset_url: uploadedImageUrl.data.url,
+              thumb_url: uploadedImageUrl.data.url
+            };
+          }
+          if (item.type === 'video') {
+            const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+            const uploadedVideoUrl = await ImageUtils.uploadImage(item.video_path);
+            return {
+              ...item,
+              asset_url: uploadedImageUrl.data.url,
+              thumb_url: uploadedImageUrl.data.url,
+              video_path: uploadedVideoUrl.data.url
+            };
+          }
+
+          return item;
+        });
+
+        const newAttachments = await Promise.all(allAttachmentPromises);
+
         response = await SignedMessageRepo.sendSignedMessage(
           selectedChannel?.id,
           message,
-          channelType
+          channelType,
+          newAttachments
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code block here uploads images and videos based on their type. However, there's a potential issue with error handling. If the upload fails for any reason (network issues, server errors, etc.), it could cause the entire Promise.all to fail. Consider adding error handling within each map iteration to ensure one failed upload doesn't stop others.

Also, there's a redundancy in lines 111 and 112 where the same image URL is uploaded twice for video type attachments. This should be corrected.

```diff
        const allAttachmentPromises: Promise<void>[] = attachments.map(async (item: any) => {
          try {
            if (item.type === 'image') {
              const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
              return {
                ...item,
                asset_url: uploadedImageUrl.data.url,
                thumb_url: uploadedImageUrl.data.url
              };
            }
            if (item.type === 'video') {
-112:               const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+112:               const uploadedThumbUrl = await ImageUtils.uploadImage(item.thumb_url);
              const uploadedVideoUrl = await ImageUtils.uploadImage(item.video_path);
              return {
                ...item,
-116:                 asset_url: uploadedImageUrl.data.url,
+116:                 asset_url: uploadedVideoUrl.data.url,
-117:                 thumb_url: uploadedImageUrl.data.url,
+117:                 thumb_url: uploadedThumbUrl.data.url,
                video_path: uploadedVideoUrl.data.url
              };
            }
          } catch (error) {
            console.error(`Failed to upload ${item.type}: `, error);
            return item;
          }

          return item;
        });

        const newAttachments = await Promise.all(allAttachmentPromises);

         response = await SignedMessageRepo.sendSignedMessage(
           selectedChannel?.id,
           message,
          channelType,
          newAttachments
        );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->